test/libsolidity/SemanticTest.cpp

```diff
@@ -92,18 +91,33 @@ SemanticTest::SemanticTest(
 		m_shouldRun = false;
 
 	auto const eofEnabled = solidity::test::CommonOptions::get().eofVersion().has_value();
-	std::string compileViaYul = m_reader.stringSetting("compileViaYul", eofEnabled ? "true" : "also");
 
-	if (compileViaYul == "false" && eofEnabled)
+	auto const compileViaYul = m_reader.enumSetting<CompileViaYul>(
+		"compileViaYul",
+		{
+			{toString(CompileViaYul::False), CompileViaYul::False},
+			{toString(CompileViaYul::True), CompileViaYul::True},
+			{toString(CompileViaYul::Also), CompileViaYul::Also},
+			{toString(CompileViaYul::OnlyOnEOF), CompileViaYul::OnlyOnEOF}
+		},
+		eofEnabled ? toString(CompileViaYul::True) : toString(CompileViaYul::Also)
+	);
+
+	solUnimplementedAssert(compileViaYul != CompileViaYul::OnlyOnEOF,
+		"'compileViaYul: onlyOnEOF' is not yet supported in semantic tests.");
```

Comment:
```suggestion
	solUnimplementedAssert(
		compileViaYul != CompileViaYul::OnlyOnEOF,
		"'compileViaYul: onlyOnEOF' is not yet supported in semantic tests."
	);
```

Same in `SyntaxTest.cpp`.

Reply:
Or maybe this. More compact than the above and has the nice side-effect of not being as easily confused with an assertion:
```suggestion
	if (compileViaYul == CompileViaYul::OnlyOnEOF)
		solUnimplemented("'compileViaYul: onlyOnEOF' is not yet supported in semantic tests.");
```